src/Server/CertificateReloader.cpp

```diff
@@ -65,24 +70,36 @@ int CertificateReloader::setCertificate(SSL * ssl)
 }
 
 
-void CertificateReloader::init()
+void CertificateReloader::init(MultiData * pdata, SSL_CTX * ctx)
 {
     LOG_DEBUG(log, "Initializing certificate reloader.");
 
     /// Set a callback for OpenSSL to allow get the updated cert and key.
 
-    auto* ctx = Poco::Net::SSLManager::instance().defaultServerContext()->sslContext();
-    SSL_CTX_set_cert_cb(ctx, callSetCertificate, nullptr);
-    init_was_not_made = false;
+    SSL_CTX_set_cert_cb(ctx, callSetCertificate, reinterpret_cast<void *>(pdata));
```

Comment:
MultiData object must not moved after creation, so pointer can be passed as callback parameter.